### ISSUE TITLE
Cannot install into existing React Native project start from React Native seed

### ISSUE THREAD

**Issue by biapar** [GeekyAnts/NativeBase#5308]
### Description

I try to install into existing RN project start from React Native seed

### CodeSandbox/Snack link

N/A

### Steps to reproduce

Hi, I try to install into existing RN project start from React Native seed.
```
    "react": "18.0.0",
    "react-native": "0.69.4",
```
I've this compatibility error:

<img width="709" alt="image" src="https://user-images.githubusercontent.com/661517/187146027-b47f6507-18ec-4baa-b256-accd55550e5e.png">

### NativeBase Version

3.4.13

### Platform

- [X] Android
- [ ] CRA
- [ ] Expo
- [X] iOS
- [ ] Next

**Comment by leighton-tidwell** [comment]
I am also getting this exact same issue. The same steps to recreate it as well.

**Comment by biapar** [comment]
> I am also getting this exact same issue. The same steps to recreate it as well.

Have you any news?

**Comment by leighton-tidwell** [comment]
> > I am also getting this exact same issue. The same steps to recreate it as well.
> 
> Have you any news?

No news, also figured out that if you `npm -i --force` it will break everything.

**Comment by biapar** [comment]
> > > I am also getting this exact same issue. The same steps to recreate it as well.
> > 
> > 
> > Have you any news?
> 
> No news, also figured out that if you `npm -i --force` it will break everything.

No. Don't use the force option.

**Comment by ankit-tailor** [comment]
Hey, Thanks for reporting the issue. We'll look into it.

**Comment by rohankleem** [comment]
I reported this 18 days ago, and has since been unresolved https://github.com/GeekyAnts/NativeBase/issues/5259#issuecomment-1214278468

**Comment by Shahzad6077** [comment]
Hey @biapar, any updates? or any workaround for this issue

**Comment by biapar** [comment]
> Hi all, use `--legacy-peer-deps`

Is't a workaround or fix?

**Comment by MrVico** [comment]
I wouldn't close this issue, using that command does not solve the problem!

**Comment by MrVico** [comment]
Hi @ankit-tailor ,
Do you have any news for us about this issue?
I don't think that using the `--legacy-peer-deps` should be the solution for this as it also makes a bunch of warnings emerge in the console when doing a simple `npm i`!

**Comment by turnerguo** [comment]
https://nativebase.io/roadmap
<img width="306" alt="image" src="https://user-images.githubusercontent.com/633880/189288078-3d936c7f-fefb-44ad-9786-bb5cb6c03dd1.png">

Looks like the latest expo SDK are not officially support yet.

**Comment by biapar** [comment]
Also the last RN version.

**Comment by evulse** [comment]
Might help someone but I just installed react-dom to the expected version prior to install to avoid it trying to install 18.2.0

```npm i react-dom@18.0.0 ```

**Comment by MrVico** [comment]
@evulse That allowed you to run `npm i native-base` without getting errors?

I still get the same error which seems normal as it's trying to install the highest version of react dom due to the `^`

> npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.2.0" from react-dom@18.2.0
npm ERR! node_modules/react-dom
npm ERR!   react-dom@"^18.0.0" from the root project
npm ERR!   peer react-dom@"*" from native-base@3.4.15
npm ERR!   node_modules/native-base
npm ERR!     native-base@"*" from the root project

Am I missing something here?

**Comment by evulse** [comment]
@MrVico It looks like it's still trying to install `18.2.0` maybe because it's already installed or maybe because it's just because your package file has `^18.0.0` 

Just go in to your package.json and change it from `^18.0.0` to `18.0.0` to force the version

**Comment by MrVico** [comment]
@evulse You are right, don't know what I did but after using your command I could run `npm i native-base` without errors (although a lot of warnings now surface when running `npm i` but at least I can use native base now, thanks!)

**Comment by nabinem** [comment]
getting same error when i do npm i native-base on new react native 0.70
but i can install with flag --legacy-peer-deps
but I wonder if native base is fully compatible with react 18 and react native 0.70 and I don't get any issues in future so I dont have to revert back to older version of react native and react

**Comment by biapar** [comment]
So, It's ok for RN0.70+ / React 18+?

**Comment by JBertling** [comment]
Can confirm that I ran into this problem and @evulse's suggestion got things running for me too. Appreciate the tip! As mentioned before I received numerous warnings regarding peer dependencies, but it allowed me to run it via npm without erroring out and I've verified that I'm seeing components working.

**Comment by itzmatheus** [comment]
[Stackoverflow](https://es.stackoverflow.com/questions/565638/error-al-instalar-native-base/565770?noredirect=1#comment1000424_565770)
1. Add overrides react v18.
```
  "overrides": {
    "react": "18.0.0"
  },
```
  
2. Then run 

> npm i native-base

**Works for me.**
PS: When i use --force flag to install native-base the emulator crashes after app refresh many times without edit the code.

![image](https://user-images.githubusercontent.com/31101192/199597290-fe6e4ae4-559e-4f8f-a2bf-427205761471.png)

**Comment by stivenm16** [comment]
> [Stackoverflow](https://es.stackoverflow.com/questions/565638/error-al-instalar-native-base/565770?noredirect=1#comment1000424_565770)
> 
> @biapar @leighton-tidwell
> 
> 1. Add overrides react v18.
> 
> ```
>   "overrides": {
>     "react": "18.0.0"
>   },
> ```
> 
> 2. Then run
> 
> > npm i native-base
> 
> **Works for me.** PS: When i use --force flag to install native-base the emulator crashes after app refresh many times without edit the code.
> 
> ![image](https://user-images.githubusercontent.com/31101192/199597290-fe6e4ae4-559e-4f8f-a2bf-427205761471.png)

I added --legacy-peer-deps flag too and it works fine. I'm using node v16.13.0 as someone suggests above

**Comment by serdarde** [comment]
If you get this error during the Expo's EAS build, you can solve the problem with adding EAS hooks in your package.json file.

Add his line in the Scripts tag ` "eas-build-pre-install": "npm config set legacy-peer-deps true"`

Source: [https://stackoverflow.com/a/74268002/1981420](https://stackoverflow.com/a/74268002/1981420)
